torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -212,24 +273,13 @@ def hook_with_zero_fn(
                 gradient bucket.
         """
         fut = hook(state, bucket)
-        overlap_info = zero._overlap_info
-        bucket_index = bucket.index()
-
-        # Proceed as normal until the DDP buckets have been rebuilt
-        if not ddp_ref()._has_rebuilt_buckets:  # type: ignore[union-attr]
-            assert overlap_info.status == _OverlapStatus.UNINITIALIZED
+        if _hook_with_zero_step_setup(ddp_ref, zero, bucket) == _HookDirective.RETURN:
```

Comment:
OOC, why do we need a new `_HookDirective` type, instead of directly checking whether the `_OverlapStatus` state is `INITIALIZED`?